common/djangoapps/third_party_auth/views.py

```diff
@@ -110,3 +112,14 @@ def post_to_custom_auth_form(request):
         'hmac': pipeline_data['hmac'],
     }
     return render(request, 'third_party_auth/post_custom_auth_entry.html', data)
+
+
+class IdPRedirectView(View):
+    def get(self, request, *args, **kwargs):
+        next_destination_url = get_next_url_for_login_page(request)
+
+        try:
+            url = pipeline.get_login_url(kwargs['provider_slug'], pipeline.AUTH_ENTRY_LOGIN, next_destination_url)
```

Comment:
Note: In order to successfully redirect, this assumes that the next destination is whitelisted [here](https://github.com/edx/edx-internal/blob/ee25ea45033941a8f3393d2fc7e1d4de5a1d3684/ansible/vars/prod-edx.yml#L299). If you're testing this locally, you will have to update your `devstack_docker.py` file to add whatever URL you are redirecting to to the `LOGIN_REDIRECT_WHITELIST` setting.

Reply:
That might be good to put in a comment.

Reply:
Done! See line 142.